metomi/rose/host_select_client.py

```diff
@@ -17,19 +15,38 @@
 # You should have received a copy of the GNU General Public License
 # along with Rose. If not, see <http://www.gnu.org/licenses/>.
 # -----------------------------------------------------------------------------
-
+import json
 import sys
 
-from subprocess import call
+import psutil
 
 
 def main():
-    command = ['etc/bin/rose-test-battery', '-j', '5']
-    if call(command + ['--state=save']):
-        # Non-zero return code
-        sys.stderr.write('\n\nRerunning Failed Tests...\n\n')
-        # Exit with final return code
-        sys.exit(call(command + ['--state=save,failed', '-v']))
+    # read metrics from stdin
+    line = True
+    metrics = ''
+    while True:
+        line = sys.stdin.readline().strip()
+        if '**start**' in line:
+            continue
+        if '**end**' in line:
+            break
+        metrics += f'\n{line}'
```

Comment:
Made it so any junk before the start marker is ignored, added a test.

On the infinite loop thing, yes you could end up with an infinite loop here, however, we can't fix this because we are reading from stdin.

`while line` would just break on the first blank line and would still get into the infinite loop because `sys.stdin.readline` will hang whilst waiting for something to read.